src/ninshiki_cpp/detector/lbp_detector.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2021 ICHIRO ITS
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include <map>
+#include <string>
+#include <vector>
+#include <opencv2/imgproc.hpp>
+
+// #include "detector.cpp"
+#include "ninshiki_cpp/detector/lbp_detector.hpp"
+
+namespace ninshiki_cpp
+{
+namespace detector
+{
+
+// LBPDetector::LBPDetector() : Detector() {}
+
+LBPDetector::LBPDetector()
+{
+    config_path = "../../data/" + utils::get_host_name() + "lbp_classifier/ball_cascade.xml";
+    classifier_loaded = loadClassifier(config_path);
+}
+
+bool LBPDetector::loadClassifier(std::string config_path) {
+    if (!cascade_detector_.load(config_path))
+    {
+        printf("failed to load cascade %s\n", config_path);
+        return false;
+    }
+
+    return true;
+}
+
+void LBPDetector::detection(cv::Mat input)
+{
+    img_width = static_cast<double>(input.cols);
+    img_height = static_cast<double>(input.rows);
+
+    std::vector<cv::Rect> rects;
+    if (!classifier_loaded)
+        return;
+
+    cv::Mat gray = input.clone();
+    cv::cvtColor(input, gray, cv::COLOR_BGR2GRAY);
```

Comment:
please do this instead, prevent unnecessary copy of the `input` variable before conversion.
```c++
cv::Mat gray;
cv::cvtColor(input, gray, ...);
```